sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/QueueReceiverAsyncClient.java

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.amqp.AmqpRetryOptions;
+import com.azure.core.amqp.AmqpRetryPolicy;
+
+import com.azure.core.amqp.implementation.AmqpReceiveLink;
+import com.azure.core.amqp.implementation.MessageSerializer;
+import com.azure.core.amqp.implementation.RetryUtil;
+import com.azure.core.amqp.implementation.TracerProvider;
+
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.servicebus.implementation.ServiceBusConnectionProcessor;
+import com.azure.messaging.servicebus.implementation.ServiceBusReceiveLinkProcessor;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.SignalType;
+
+import java.io.Closeable;
+import java.time.Instant;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Supplier;
+
+import static com.azure.core.util.FluxUtil.fluxError;
+
+/**
+ * An <b>asynchronous</b> receiver responsible for reading {@link Message} from either a specific Queue.
+ *
+ */
+@ServiceClient(builder = QueueClientBuilder.class, isAsync = true)
+public final class QueueReceiverAsyncClient implements Closeable {
+
+    private static final String RECEIVER_ENTITY_PATH_FORMAT = "%s";
+
+    private final AtomicBoolean isDisposed = new AtomicBoolean();
+    private final ClientLogger logger = new ClientLogger(QueueReceiverAsyncClient.class);
+    private final String fullyQualifiedNamespace;
+    private final String queueName;
+    private final ServiceBusConnectionProcessor connectionProcessor;
+    private final MessageSerializer messageSerializer;
+    private final int prefetchCount;
+    private final boolean isSharedConnection;
+    private final TracerProvider tracerProvider;
+    private final ReceiveMode defaultReceiveMode = ReceiveMode.PEEKLOCK;
+
+
+    AmqpRetryOptions defaultRetryOptions =  new AmqpRetryOptions();
+
+    /**
+     * Keeps track of the open consumers keyed by linkName. The link name is generated as: {@code
+     * "partitionId_GUID"}. For receiving from all partitions, links are prefixed with {@code "all-GUID-partitionId"}.
+     */
+    private final ConcurrentHashMap<String, ServiceBusAsyncConsumer> openConsumers =
```

Comment:
The concept of partitions doesn't exist here. This class was introduced so that one consumer could listen to multiple partitions in Event Hubs. What you're copying is the incorrect class.

Reply:
We have multiple session , similar concept and  we could use it for that purpose.

Reply:
It wouldn't be structured in this way and be returned from the .receive() function. A QueueReceiver would be tied to a single `ServiceBusAsyncConsumer`. You'd probably have another class SessionReceiver or something internally.